interview-questions.md

```diff
@@ -8,36 +8,91 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. As a developer, I am creating a Rails application with a model called Cohort and a model called Student. The cohort model has_many students, but OOPS! I forgot to add the foreign key. How can I fix this mistake? What is the name of the foreign key? Would the foreign key be on the Cohort model or the Student model?
 
-Your answer:
-
-Researched answer:
+Your answer: Adding a migration file add_collumn named cohort_id to add the foreign key to the Student model since each cohort will have assigned many students. Since the Cohort model will have the has_many, the Student model must have belongs_to to finalize the associated.  
+
+Researched answer: 
+Adding a column: $rails generate migration add_collumn_to_student
+Make the following changes in db/migrate file def change
+
+```rb 
+Class AddColumnToStudent < ActiveRecord::Migration[6.0]
+	def change
+		Add_column :students, :cohort_id, :integer 
+	end
+end
+```
+Save file/ run: $ rails db:migrate
+Update the Class to create the Association
+app/models/cohort.rb
+```rb
+class Cohort < ApplicationRecord
+	has_many :students
+end
+```
+app/models/student.rb
+```rb
+class Student < ApplicationRecord
+	belongs_to  :cohort
+end
+```
 
 2. Which RESTful routes must always be passed params? Why?
 
-Your answer:
+Your answer: I completely drew a blank on this, so I will go straight into the research to help re-spark my memory. 
+
+Researched answer:  The RESTful routes that must have a param passed as well are: Show, New, Create, Update, and Delete. This is because when each of these controller methods is called into action to perform tasks such as manipulating, structuring, displaying or destructing specific pieces of data.
```

Comment:
I would review RESTful routes; the picture in the syllabus in the RESTful routes lesson is a great tool